vault/resource_consul_secret_backend_role.go

```diff
@@ -189,17 +203,17 @@ func consulSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) erro
 	log.Printf("[DEBUG] Configuring Consul secrets backend role at %q", path)
 
 	if _, err := client.Logical().Write(path, data); err != nil {
-		return fmt.Errorf("error writing role configuration for %q: %s", path, err)
+		return diag.Errorf("error writing role configuration for %q: %s", path, err)
 	}
 
 	d.SetId(path)
-	return consulSecretBackendRoleRead(d, meta)
+	return consulSecretBackendRoleRead(ctx, d, meta)
 }
 
-func consulSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
-	client, e := provider.GetClient(d, meta)
-	if e != nil {
-		return e
+func consulSecretBackendRoleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
```

Comment:
Would you mind reverting this to be `client, e` ? It will make it consistent with all the other functions.

Reply:
Sure I don't mind. I thought it was just an overlooked variable name since both e and err were used.